src/roswire/common/package.py

```diff
@@ -237,9 +192,14 @@ def __iter__(self) -> Iterator[str]:
         """
         yield from self.__contents
 
-    @staticmethod
-    def from_dict(d: List[Dict[str, Any]]) -> 'PackageDatabase':
-        return PackageDatabase((Package.from_dict(dd) for dd in d), [])
+    @classmethod
+    def _from_dict_internal(cls, d: List[Dict[str, Any]]) -> 'PackageDatabase':
+        return cls([Package.from_dict(dd) for dd in d], [])
```

Comment:
Same as above. Assumes implementation knowledge of the constructor.  `for_packages_and_paths` can do the job here.`